src/garage/tf/algos/rl2.py

```diff
@@ -483,23 +480,30 @@ def _concatenate_paths(self, paths):
                 values of shape :math:`[max_episode_length, S^*]`
 
         """
-        observations = np.concatenate([path['observations'] for path in paths])
+        env_infos = {
+            k: np.concatenate([b.env_infos[k] for b in episode_list])
+            for k in episode_list[0].env_infos.keys()
+        }
+        agent_infos = {
+            k: np.concatenate([b.agent_infos[k] for b in episode_list])
+            for k in episode_list[0].agent_infos.keys()
+        }
```

Comment:
I think there might be a utility in tensor utils for concatenating all arrays in a dictionary. If not, it would probably be a good idea to create one.

Reply:
I find a function called concat_tensor_dict_list. But it seems not suitable to use here. I think the code here is not generous as it is a modification from EpisodeBatch.concatenate, so I will probably just leave it there.